src/cascadia/TerminalApp/AppLogic.cpp

```diff
@@ -1468,97 +652,42 @@ namespace winrt::TerminalApp::implementation
-    //   message. If there were no errors, this message will be blank.
-    // - If the user requested help on any command (using --help), this will
-    //   contain the help message.
-    // - If the user requested the version number (using --version), this will
-    //   contain the version string.
-    // Arguments:
-    // - <none>
-    // Return Value:
-    // - the help text or error message for the provided commandline, if one
-    //   exists, otherwise the empty string.
-    winrt::hstring AppLogic::ParseCommandlineMessage()
-    {
-        return winrt::to_hstring(_appArgs.GetExitMessage());
-    }
-
-    // Method Description:
-    // - Returns true if we should exit the application before even starting the
-    //   window. We might want to do this if we're displaying an error message or
-    //   the version string, or if we want to open the settings file.
-    // Arguments:
-    // - <none>
-    // Return Value:
-    // - true iff we should exit the application before even starting the window
-    bool AppLogic::ShouldExitEarly()
-    {
-        return _appArgs.ShouldExitEarly();
-    }
-
-    bool AppLogic::FocusMode() const
-    {
-        return _root ? _root->FocusMode() : false;
-    }
-
-    bool AppLogic::Fullscreen() const
-    {
-        return _root ? _root->Fullscreen() : false;
-    }
-
-    void AppLogic::Maximized(bool newMaximized)
+    Windows::Foundation::Collections::IMapView<Microsoft::Terminal::Control::KeyChord, Microsoft::Terminal::Settings::Model::Command> AppLogic::GlobalHotkeys()
     {
-        if (_root)
-        {
-            _root->Maximized(newMaximized);
-        }
+        return _settings.GlobalSettings().ActionMap().GlobalHotkeys();
     }
 
-    bool AppLogic::AlwaysOnTop() const
+    Microsoft::Terminal::Settings::Model::Theme AppLogic::Theme()
     {
-        return _root ? _root->AlwaysOnTop() : false;
+        return _settings.GlobalSettings().CurrentTheme();
     }
 
-    bool AppLogic::AutoHideWindow()
+    TerminalApp::TerminalWindow AppLogic::CreateNewWindow()
     {
-        if (!_loadedInitialSettings)
+        if (_settings == nullptr)
         {
-            // Load settings if we haven't already
             ReloadSettings();
         }
 
-        return _settings.GlobalSettings().AutoHideWindow();
-    }
-
-    Windows::Foundation::Collections::IMapView<Microsoft::Terminal::Control::KeyChord, Microsoft::Terminal::Settings::Model::Command> AppLogic::GlobalHotkeys()
-    {
-        return _settings.GlobalSettings().ActionMap().GlobalHotkeys();
-    }
+        auto ev = winrt::make_self<SettingsLoadEventArgs>(false,
+                                                          _settingsLoadedResult,
+                                                          _settingsLoadExceptionText,
+                                                          winrt::multi_threaded_vector<SettingsLoadWarnings>(std::move(_warnings)),

```

Comment:
You should be careful with this `move` here. I think you're moving from `_warnings` when the settings are loaded (to construct an event for every window) and then again when a new window is created (to construct an event for its creation).

The second time will be a no-op, but also it will produce an empty vector. If that is the desired behavior, it's better to make it explicit.

Reply:
Y'know, I didn't have it as a `move` origianlly (before https://github.com/microsoft/terminal/pull/14825#discussion_r1113338128, c79f27c3dac686c5f30b6448831d856a61fd57ca). It seems to always work as expected even with the `move`, but it does _feel_ wrong. 

I might go back to making a new IVector (without the `move`) just cause I don't think I trust this.